server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -375,6 +404,37 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
         }
     }
 
+    /**
+     * Initializes the search cache with a defined capacity.
+     * The capacity of the cache is based on user configuration for {@link Node#NODE_SEARCH_CACHE_SIZE_SETTING}.
+     * If the user doesn't configure the cache size, it fails if the node is a data + search node.
+     * Else it configures the size to 80% of available capacity for a dedicated search node, if not explicitly defined.
+     */
+    private void initializeRemoteStoreFileCache(Settings settings) {
+        if (DiscoveryNode.isSearchNode(settings)) {
+            long capacity = NODE_SEARCH_CACHE_SIZE_SETTING.get(settings).getBytes();
+
+            if (capacity == 0) {
+                if (DiscoveryNode.hasRole(settings, DiscoveryNodeRole.DATA_ROLE)) {
```

Comment:
Should we be more conservative here by only defaulting to 80% if the node has a single role as "search"? That would mean there is no default value for a "cluster_manager,search" configuration, for example.

Reply:
Sure. Let me update that.